push-to-s3-and-invalidate-cloudfront/action.yml

```
@@ -0,0 +1,150 @@
+      Set this variable to an empty string if you don't want to exlcude any files
+      If unspecified will default to `pr-previews/*`
+    required: false
+    default: "pr-previews/*"
+  invalidate-cloudfront-cache:
+    description: A flag to indicate if the cloudfront cache should be invalidated
+    required: false
+    default: "false"
+
+outputs:
+  dir_to_upload_to:
+    description: The S3 directory to upload the files to. Computed based on user input
+    value: ${{ steps.upload.dir_to_upload_to }}
+  dir_to_delete_from:
+    description: The S3 directory to upload the files to. Computed based on user input
+    value: ${{ steps.delete.dir_to_delete_from }}
+  files_to_exlude:
+    description: Files to exclude from S3 when using the sync command.
+    value: ${{ steps.exclude.files_to_exclude }}
+
+runs:
+  using: composite
+  steps:
+    - name: Compute the S3 directory to upload to
+      shell: bash
+      id: upload
+      run: |
+        if [[ "${{ inputs.s3-dir-to-upload-to }}" == "pr-previews" ]]
+        then
+          dir_to_upload_to="${{ inputs.s3-dir-to-upload-to }}/PR-${{ inputs.pull-request-number }}"
+        elif [[ "${{ inputs.s3-dir-to-upload-to }}" == "/" ]]
+        then
+          dir_to_upload_to=""
+        else
+          dir_to_upload_to="${{ inputs.s3-dir-to-upload-to }}"
+        fi
+        echo "dir_to_upload_to=$dir_to_upload_to" >> $GITHUB_OUTPUT
+
+    - name: Compute the S3 directory to delete from
+      shell: bash
+      id: delete
+      run: |
+        if [[ "${{ inputs.s3-dir-to-delete-from }}" == "pr-previews" ]]
+        then
+          dir_to_delete_from="${{ inputs.s3-dir-to-delete-from }}/PR-${{ inputs.pull-request-number }}"
+        elif [[ "${{ inputs.s3-dir-to-delete-from }}" == "/" ]]
+        then
+          dir_to_delete_from=""
+        else
+          dir_to_delete_from="${{ inputs.s3-dir-to-delete-from }}"
+        fi
+        echo "dir_to_delete_from=$dir_to_delete_from" >> $GITHUB_OUTPUT
+
+    - name: Compute files to exclude
+      shell: bash
+      id: exclude
+      run: |
+        files_to_exlude=""
+        if [[ "${{ inputs.s3-files-to-exclude }}" == *,* ]]
+        then
+          IFS=','
+          for i in `echo "${{ inputs.s3-files-to-exclude }}"`
+          do
+            files_to_exlude="$files_to_exlude --exclude \"$i\""
+          done
+          unset IFS
+        elif [[ "${{ inputs.s3-files-to-exclude }}" == "pr-previews" ]] || [[ "${{ inputs.s3-files-to-exclude }}" != "" ]]
+        then
+          files_to_exlude="--exclude \"${{ inputs.s3-files-to-exclude }}\""
+        fi
+        echo "files_to_exlude=$files_to_exlude" >> $GITHUB_OUTPUT
+
+    - name: Upload the files to S3
+      shell: bash
+      if: inputs.s3-action == 'upload' || (inputs.s3-action == '' && github.event_name == 'pull_request' && contains(fromJson('["opened", "synchronize"]'), github.event.action))
```

Question: do we need `fromJson` here? :thinking: 
```suggestion
      if: inputs.s3-action == 'upload' || (inputs.s3-action == '' && github.event_name == 'pull_request' && contains(['opened', 'synchronize'], github.event.action))
```

---

Yes we need it! Without `fromJson` the comparison becomes `contains(string, string)` but with `fromJson` it becomes `contains(array, string)` which is what we need 😄